### ISSUE TITLE
The header content contains invalid characters crash

### ISSUE THREAD

**Issue by arrilot** [JonathanBennett/prerender-redis-cache#17]
Steps to reproduce.

1. Everything works fine without redis cache plugin
2. After `server.use(require('prerender-redis-cache'));` the first request to a page works just fine too. I can see prerendered HTML and it is stored to redis aswell
3. The second request to this page crashes prerender nodejs process with stack trace

```
_http_outgoing.js:494
    throw new TypeError('The header content contains invalid characters');
    ^

TypeError: The header content contains invalid characters
    at validateHeader (_http_outgoing.js:494:11)
    at ServerResponse.setHeader (_http_outgoing.js:498:3)
    at Object.newRes.setHeader (/opt/prerender/lib/server.js:318:8)
    at Command.callback (/opt/prerender/node_modules/prerender-redis-cache/lib/prerenderRedisCache.js:79:29)
    at normal_reply (/opt/prerender/node_modules/redis/index.js:726:21)
    at RedisClient.return_reply (/opt/prerender/node_modules/redis/index.js:824:9)
    at JavascriptRedisParser.returnReply (/opt/prerender/node_modules/redis/index.js:192:18)
    at JavascriptRedisParser.execute (/opt/prerender/node_modules/redis-parser/lib/parser.js:553:10)
    at Socket.<anonymous> (/opt/prerender/node_modules/redis/index.js:274:27)
    at emitOne (events.js:116:13)
```

After some debugging I figured out that this happens because of Set-cookie header:
`console.log( key + ':' + headers[key]);`

```
Set-Cookie:PHPSESSID=e89f4a9d0ce8d74b4b124629e6af9ae7; path=/; HttpOnly
BITRIX_SM_SALE_UID=15802; expires=Mon, 16-Sep-2019 19:22:12 GMT; Max-Age=31104000; path=/
```

Probably line break is the case.
ATM I just use quick and dirty fix for this problem - my own cop of redis cache plugin where this header is simply skipped, 

```
if (headers.hasOwnProperty(key) && key !== 'Set-Cookie') {
      res.setHeader(key, headers[key]);
}
```

However this must be addressed in a more proper way.

My point. - If everything works fine with these headers without plugin, it should continue to work fine with the installed plugin aswell.

nodejs -v
v8.10.0

**Comment by HenkVanMaanen** [comment]
I have the same problem. @arrilot on what line and in which file did you place this? 

`
if (headers.hasOwnProperty(key) && key !== 'Set-Cookie') {
      res.setHeader(key, headers[key]);
}
`

**Comment by arrilot** [comment]
Hi, 
1. copy `node_modules/prerender-redis-cache/lib/prerenderRedisCache.js` to `lib/plugins/prerenderRedisCacheCustom.js`
2. Line 77
3. add `server.use(prerender.prerenderRedisCacheCustom());` to `server.js`

**Comment by vdt-mik** [comment]
i have same problem:
```
1. curl http://localhost:3000/render\?url\=https://www.google.com
log: app_1       | 2019-06-25T07:09:09.674Z - 0 https://play.google.com/log?format=json&hasfast=true
app_1       | 2019-06-25T07:09:10.847Z warning: unable to set header: set-cookie
app_1       | 2019-06-25T07:09:10.864Z got 200 in 3018ms for https://www.google.com/
app_1       | 2019-06-25T07:09:10.871Z error closing Chrome tab unable to dispose of browser context
2. curl http://localhost:3000/render\?url\=https://www.google.com
logs: app_1       | 2019-06-25T07:27:44.577Z getting https://www.google.com/
app_1       | _http_outgoing.js:494
app_1       |     throw new TypeError('The header content contains invalid characters');
app_1       |     ^
app_1       | 
app_1       | TypeError: The header content contains invalid characters
app_1       |     at validateHeader (_http_outgoing.js:494:11)
app_1       |     at ServerResponse.setHeader (_http_outgoing.js:498:3)
app_1       |     at Object.newRes.setHeader (/var/www/lib/server.js:318:8)
app_1       |     at Command.callback (/var/www/lib/plugins/prerenderRedisCacheCustom.js:78:29)
app_1       |     at normal_reply (/var/www/node_modules/redis/index.js:726:21)
app_1       |     at RedisClient.return_reply (/var/www/node_modules/redis/index.js:824:9)
app_1       |     at JavascriptRedisParser.returnReply (/var/www/node_modules/redis/index.js:192:18)
app_1       |     at JavascriptRedisParser.execute (/var/www/node_modules/redis-parser/lib/parser.js:553:10)
app_1       |     at Socket.<anonymous> (/var/www/node_modules/redis/index.js:274:27)
app_1       |     at emitOne (events.js:116:13)
app_1 exited with code 1
```
i tried to use your fix, but it doesn`t work too

**Comment by thoop** [comment]
It looks like there might be another error in your logs that might be causing an issue:
```
app_1       | 2019-06-25T07:09:10.871Z error closing Chrome tab unable to dispose of browser context
```
Can you confirm the content being returned looks correct? Or are you saving incorrect content into redis due to some error with the Prerender server?

**Comment by jaynetics** [comment]
i've made a fork that ignores all invalid headers: https://github.com/jaynetics/prerender-redis-cache (just like the prender server [ignores them](https://github.com/prerender/prerender/blob/c410ad0abff0f19aa5f817be61bb39cf73c50125/lib/server.js#L365-L369))

**Comment by JonathanBennett** [comment]
Happy to merge in a fix if you have one that works @jaynetics. I'd like to resolve this issue in the plugin if possible, I just haven't had time to investigate the fix myself. 👌

**Comment by jaynetics** [comment]
@JonathanBennett whoops, i wasn't sure this is still maintained. i've just opened a PR with my fix, which removes invalid headers using the same regexp that node uses to throw the error above. tbh i didn't investigate the root cause. i guess the bug is actually in https://www.github.com/prerender/prerender and just isn't noticeable with other plugins like prerender-memory-cache because they don't forward headers. either way, i guess it can't do much damage to drop invalid ones.

**Comment by JonathanBennett** [comment]
This is now fixed, merged, and published as 0.2.2. Thanks so much for the PR @jaynetics!
